### DISCUSSION TITLE
SSH and HTTPS binary cache sources

### DISCUSSION THREAD

**Discussion by Be-ing** [microsoft/vcpkg#20670] [Ideas]
****

**Is your feature request related to a problem? Please describe.**
Currently binary caching can only use two types of sources, local files and NuGet servers. The NuGet server software only runs on Windows which limits options for projects to host their own vcpkg binary cache server.

**Proposed solution**
Allow specifying HTTPS and SSH servers in VCPKG_BINARY_SOURCES. SSH could be used for read-write access on CI while developers could download the cached packages via HTTPS as a read-only source.

**Describe alternatives you've considered**
FTP could be implemented but that's not used very much anymore (FWIW Firefox recently removed FTP support).

**Additional context**
Mounting SSH servers as filesystems can be done which is usable with the generic file caching in vcpkg. However the tools to do this are not common on Windows or macOS, which makes it cumbersome or impossible to use this option in some use cases, for example GitHub Actions.

**Comment by cenit** [comment]
there are many nuget servers running as a container on linux if you need a fast alternative.

look for example at BaGet

**Comment by Be-ing** [comment]
BaGet is an option but it requires the .NET SDK which is not packaged in many Linux distributions. I know Microsoft has their own APT repository for it, but I think many people would be uncomfortable running third party binary packages on their server.
